lib/Conversion/TritonGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -126,19 +126,18 @@ struct LoadOpConversion
       for (size_t wordIdx = 0; wordIdx < nWords; ++wordIdx) {
         for (size_t wordElem = 0; wordElem < wordNElems; ++wordElem) {
           size_t elemOffset = vecStart + wordIdx * wordNElems + wordElem;
-          Value otherVal = other ? otherElems[elemOffset] : zeroVal;
-          auto *curBlock = rewriter.getBlock();
-          auto *endBlock = curBlock->splitBlock(rewriter.getInsertionPoint());
-          endBlock->addArgument({valueElemTy}, {loc});
-          auto *loadBlock = rewriter.createBlock(
-              curBlock->getParent(), std::next(Region::iterator(curBlock)));
-          rewriter.setInsertionPointToEnd(curBlock);
-          rewriter.create<LLVM::CondBrOp>(loc, pred, loadBlock, endBlock, otherVal);
-          rewriter.setInsertionPointToEnd(loadBlock);
-          Value loadVal = load(ptrElems[elemOffset]);
-          rewriter.create<LLVM::BrOp>(loc, loadVal, endBlock);
-          rewriter.setInsertionPointToStart(endBlock);
-          loadedVals.push_back(endBlock->getArgument(0));
+          auto loaded = rewriter.create<scf::IfOp>(loc, TypeRange({valueElemTy}), pred,
+                                     [&](OpBuilder &builder, Location loc){
+                                       auto loadVal = builder.create<LLVM::LoadOp>(loc, ptrElems[elemOffset]);
+                                       builder.create<mlir::scf::YieldOp>(loc, ValueRange({loadVal}));
+                                     },
+                                     [&](OpBuilder &builder, Location loc){
+                                       Value zeroVal = bitcast(int_val(valueElemNbits, 0), valueElemTy);
```

Comment:
i don't think this zeroVal statement is needed since it is already defined before the nested for loop.